### PR TITLE
New unit testing infrastructure 

### PR DESCRIPTION
**Pull request by densh**
This pr introduces dead simple unit testing infrastructure that's inspired by `FunSuite` from scalatest, only way way simpler. Every test suite is object that extends `tests.Suite` and contains tests defined within `test("name") { body }`. Currently there is no automatic tests discovery and every suite needs to be added manually to `tests.Main` list of known test suites. 

/cc @frgomes @gurinderu

### CODE REVIEW

File: unit-tests/src/main/scala/java/util/RandomSuite.scala [hunk 1]
```diff
@@ -212,15 +191,39 @@ object RandomTest {
...
+                   -9,
+                   11,
+                   57,
+                   -118,
+                   40,
+                   -74,
+                   -86,
+                   78,
+                   123,
+                   58))
```
Comment (densh) [inline unit-tests/src/main/scala/java/util/RandomSuite.scala]: Scalafmt please stop what are you doing to my code. @olafurpg

Comment (olafurpg) [inline unit-tests/src/main/scala/java/util/RandomSuite.scala]:
```
// format: off

// format: on
```
https://olafurpg.github.io/scalafmt/#//format:off

File: unit-tests/src/main/scala/tests/Suite.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+abstract class Suite {
+  private val tests = new mutable.UnrolledBuffer[Test]
+
+  def assert(cond: Boolean): Unit =
+    if (!cond) throw AssertionFailed else ()
+
+  def assertNot(cond: Boolean): Unit =
+    if (cond) throw AssertionFailed else ()
+
+  def assertThrows(f: => Unit): Unit =
```
Comment (olafurpg) [inline unit-tests/src/main/scala/tests/Suite.scala]: You might want to be more specific about which type of exception you expected.

File: unit-tests/src/main/scala/java/util/RandomSuite.scala (hunk 1 above)
Comment (densh) [inline unit-tests/src/main/scala/java/util/RandomSuite.scala]: I know, but it still hurts.

File: unit-tests/src/main/scala/tests/Suite.scala (hunk 2 above)
Comment (densh) [inline unit-tests/src/main/scala/tests/Suite.scala]: Good point.

**Comment by frgomes** [comment]
May I suggest utest instead of ScalaTest?
The main reason is that uTest is potentially very simple to compile with SN.
Then, in future (yes... maybe far future... but anyway...), SN codebase itself will be ready to be compiled by SN.

This is an example of how tests can be written. Deadly simple. No frills. Just asserts plain old asserts.
https://github.com/frgomes/exercise-checkout/blob/master/src/test/scala/info/rgomes/exercise/checkout/ShoppingCartSpec.scala

**Comment by frgomes** [comment]
This would be possibly enough for automagically discovering tests, but using ``uTest`` in particular:
```scala
libraryDependencies += "com.lihaoyi" %% "utest" % "0.4.3" % "test"
testFrameworks += new TestFramework("utest.runner.Framework")
```

**Comment by densh** [comment]
@frgomes 

These tests look like ScalaTest but they are in fact our own barebones testing framework. It will let us bootstrap our testing infrastructure before we're able to support something more full-featured like JUnit in Scala.js

We can't use any of the existing 3-rd party frameworks to test the core of Scala Native as that'll complicate the release procedure: to release Scala Native you need to run the tests, to run the tests you need to compile 3-rd party testing framework, to compile 3-rd-party framework you need to have a stable release of Scala Native.

**Pull request merged**